src/lib/render-lit-html.ts

```diff
@@ -384,8 +409,15 @@ const getTemplate = (result: TemplateResult) => {
   return t;
 };
 
+type RenderableCustomElement = HTMLElement & {
+  new(): RenderableCustomElement;
+  connectedCallback?(): void;
+  ssrRenderChildren?: IterableIterator<string> | undefined;
```

Comment:
I'm not sure about this. Seems like it's replacing the ElementRenderer with more responsibility on the instance, but I do like the separation of concerns with the server-only ElementRenderer interface.